Source/GTMSessionFetcher.h

```diff
@@ -731,6 +744,14 @@ NSData *_Nullable GTMDataFromInputStream(NSInputStream *inputStream, NSError **o
 // affect a fetch after the fetch has begun.
 - (void)setRequestValue:(nullable NSString *)value forHTTPHeaderField:(NSString *)field;
 
+// Holds a weak reference to `decorator`. Each registered `decorator` can add HTTP header(s) to the
+// request before it starts.  If multiple decorators add the same header to a request, the most
+// recent decorator passed to this method wins.
+- (void)addHeaderDecorator:(id<GTMFetcherHeaderDecoratorProtocol>)decorator;
```

Comment:
Is this useful to add on a per-fetcher basis? When creating an individual, standalone fetcher it would be sufficient to directly add the desired headers. I would think the main utility of the decorator would be when used with the service, where we want all fetchers created with the service to use the decorators.

From the API surface of a GTMSessionFetcher instance, I wonder if the only thing we might want to expose would be a way to explicitly skip using the service's header decorator (if, for some reason, this was desired on an individual fetcher?).

This would allow us not to need to pass decorators from the service to the individual fetcher.

Reply:
That's true, the fetcher can just reach into the service directly. Thanks, will fix.